### ISSUE TITLE
How to optimize point radiuses

### ISSUE THREAD

**Issue by Kiord** [facebookresearch/pytorch3d#1562]
Hello, thank you for this powerful library !

I am trying to optimize a point cloud positions AND radiuses with differentiable rendering (with a `PointsRenderer`).
The positions are correctly optimized but the radiuses remain the same at each iteration, even though they have strong impact on the rendered images.

I use this personalized rasterizer to directly pass the radiuses to the renderer :

```python 
class MyPointsRasterizer(PointsRasterizer):
    def forward(self, point_clouds, **kwargs) -> PointFragments:
        points_proj = self.transform(point_clouds, **kwargs)
        # get radius from the kwargs
        radius = kwargs.get("radius", self.raster_settings.radius)
        idx, zbuf, dists2 = rasterize_points(
            points_proj,
            image_size=self.raster_settings.image_size,
            radius=radius,
            points_per_pixel=self.raster_settings.points_per_pixel,
            bin_size=self.raster_settings.bin_size,
            max_points_per_bin=self.raster_settings.max_points_per_bin,
        )
        return PointFragments(idx=idx, zbuf=zbuf, dists=dists2)
  
   # create a renderer with a MyPointsRasterizer instance
   renderer_points = PointsRenderer(
        rasterizer=MyPointsRasterizer(cameras=cameras, raster_settings=raster_settings),
        compositor=AlphaCompositor(background_color=(0)),
    )
   point_clouds = ...
   radiuses = ...

   # Pass the radiuses to be optimized to the renderer call
   images = render_points(point_clouds, radius=radiuses)
```

Does the `PointsRenderer` allow for per point radius backpropagation ? If so, how ?

**Comment by Kiord** [comment]
It appears that the point positions is the only data for which the gradient is computed :
https://github.com/facebookresearch/pytorch3d/blob/f68371d3988cab39e1730281f97e91b62887b2a6/pytorch3d/renderer/points/rasterize_points.py#L217-L240

However with the Pulsar backend, the gradient of the radiuses and other fields seems to be computed :
https://github.com/facebookresearch/pytorch3d/blob/f68371d3988cab39e1730281f97e91b62887b2a6/pytorch3d/renderer/points/rasterize_points.py#L217-L240

I will try to do my stuff with `PulsarPointsRenderer` instead of `PointsRenderer`. Closing this issue.
